src/Refinements/Expr.class.st

```diff
@@ -81,21 +82,21 @@ Expr >> - rhs [
 { #category : #'theory symbols' }
 Expr >> < rhs [
 	^ECst
-		expr: (EMessageSend of: (MessageSend receiver: self selector: #< argument: rhs))
+		expr: (EBin bop: #< left: self right: rhs)
```

Comment:
Likewise.

Reply:
And same with this.  Actually there is a much bigger problem here, we treat interpreted theory symbols in an ad-hoc fashion; we get away with it for now (we only have one, Set_Set) but this doesn't scale, see https://github.com/ucsd-progsys/liquid-fixpoint/blob/develop/src/Language/Fixpoint/Smt/Theories.hs
I have been meaning to open an Issue for it the past couple of days, but frankly I want to first merge Fixpoint-over-Datatypes today.